Client/Frontend/Browser/Onboarding/OnboardingNavigationController.swift

```diff
@@ -53,13 +65,25 @@ class OnboardingNavigationController: UINavigationController {
                 return OnboardingSearchEnginesViewController(profile: profile, theme: theme)
             case .shieldsInfo:
                 return OnboardingShieldsViewController(profile: profile, theme: theme)
+            case .rewardsInfo:
+                return OnboardingRewardsViewController(profile: profile, theme: theme)
+            case .rewardsAgreement:
+                return OnboardingRewardsAgreementViewController(profile: profile, theme: theme)
+            case .adsCountdown:
+                return OnboardingAdsCountdownViewController(profile: profile, theme: theme)
+            case .adsInfo:
+                return OnboardingAdsFinishedViewController(profile: profile, theme: theme)
             }
         }
         
         var type: OnboardingViewController.Type {
             switch self {
             case .searchEnginePicker: return OnboardingSearchEnginesViewController.self
             case .shieldsInfo: return OnboardingShieldsViewController.self
+            case .rewardsInfo: return OnboardingRewardsViewController.self
+            case .rewardsAgreement: return OnboardingRewardsAgreementViewController.self
+            case .adsCountdown: return OnboardingAdsCountdownViewController.self
```

Comment:
I think as per spec we shouldn't be showing the ads stuff unless the user can actually enable ads (`BraveAds.isSupportedRegion(Locale.current.identifier)` returning true)

Reply:
Fixed.